pennylane/ops/op_math/exp.py

```diff
@@ -284,6 +283,28 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             coeffs = [c * coeff for c in coeffs]
             return self._trotter_decomposition(ops, coeffs)
 
+        if not qml.math.is_abstract(coeff) and not qml.math.isclose(qml.math.real(coeff), 0):
```

Comment:
Should the same check also be added for `Exp.generator`?

Reply:
I added the check to `Exp.has_generator` and used it in `Exp.generator`